SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -118,24 +124,32 @@ bool UReferenceCountedLockingPolicy::IsLocked(const AActor* Actor) const
 		UE_LOG(LogReferenceCountedLockingPolicy, Warning, TEXT("IsLocked called for nullptr"));
 		return false;
 	}
-	return ActorToLockingState.Contains(Actor);
-}
 
-void UReferenceCountedLockingPolicy::OnLockedActorDeleted(AActor* DestroyedActor)
-{
-	TArray<ActorLockToken> TokensToRemove;
-	for (const auto& KeyValuePair : TokenToNameAndActor)
+	// Is this Actor explicitly locked or on a locked hierarchy ownership path.
+	if (IsExplicitlyLocked(Actor) || IsOnLockedHierarchyPath(Actor))
 	{
-		if (KeyValuePair.Value.Actor == DestroyedActor)
-		{
-			TokensToRemove.Add(KeyValuePair.Key);
-		}
+		return true;
 	}
-	for (const auto& Token : TokensToRemove)
+
+	// Is the hierarchy root of this Actor explicitly locked or on a locked hierarchy ownership path.
+	TArray<AActor*> OwnershipHierarchyPath = SpatialGDK::GetOwnershipHierarchyPath(Actor);
```

Comment:
Could just use `GetTypedOuter` here

Reply:
Need our own implementation to be able to stop iterating at Actors in their OnDestroyed callback